docs/en/interfaces/cli.md

```diff
@@ -158,6 +158,116 @@ $ clickhouse-client --param_tuple_in_tuple="(10, ('dt', 10))" -q "SELECT * FROM
+If port is not specified, the default port will be used (9000).
+If database is not specified, the default database will be used.
+
+User, password, and database can be specified in the connection string either in `--user`, `--password`, `--database` command line options.
+
+The connection string must be specified in the first argument of clickhouse-client. The connection string can be combined with other [command-line-options](#command-line-options) except `--host(h)` and `--port`.
+
+### Multiple hosts {#connection_string_multiple_hosts}
+
+URI allows multiple hosts to be connected to, and the client will try to connect to those hosts using the order from URI and command line options. The hosts and ports in the URI accept comma-separated lists of values.
+
+If more than one host is supplied, or if a single host name is translated to more than one address, each host and address will be attempted one at a time until one is successful. The remaining hosts after successful connection in the list are not tried.
+
+### Percent encoding {#connection_string_uri_percent_encoding}
+
+Hosts, user name, password, database, and query parameters should be [Percent-Encoded](https://en.wikipedia.org/wiki/URL_encoding) if values contain invalid URI characters.
+
+### Examples {#connection_string_examples}
+
+Connect to localhost using port 9000 and execute the query "SELECT 1".
+
+``` bash
+clickhouse-client "clickhouse://localhost:9000" --query "SELECT 1"
+```
+
+Connect to localhost using port 9000 in interactive, multiline mode.
+
+``` bash
+clickhouse-client "clickhouse://localhost:9000" -m
+```
+
+Connect to localhost using port 9000 in interactive mode with the user specified in `--user` option.
+
+``` bash
+clickhouse-client "clickhouse://localhost:9000" --user default
+```
+
+Connect to localhost using port 9000 in interactive mode to `my_database` database specified in the command line option.
+
+``` bash
+clickhouse-client "clickhouse://localhost:9000" --database my_database
+```
+
+Connect to localhost using port 9000 in interactive mode to `my_database` database specified in the connection string.
+
+``` bash
+clickhouse-client "clickhouse://localhost:9000/my_database"
+```
+
+Connect to localhost using port 9000 in interactive mode to `my_database` database specified in the connection string and a secure connection using shorthanded 's' URI parameter.
+
+```bash
+clickhouse-client "clickhouse://localhost/my_database?s"
+```
+
+Connect to default host using default port, default user, and default database.
+
+``` bash
+clickhouse-client "clickhouse:"
+```
+
+Connect to the default host using the default port, using user user_name and no password.
+
+``` bash
+clickhouse-client "clickhouse://user_name@"
+```
+
+Connect to localhost using email as the user name. `@` symbol is percent encoded to `%40`.
+
+``` bash
+clickhouse-client "clickhouse://some_user%40some_mail.com@localhost:9000"
+```
+
```

Comment:
Bonus points for an example with > 1 host.

Reply:
Done.